### PR TITLE
email alerts sent to respondents: include session instructions #5768

### LINKED ISSUE #5768: email alerts sent to respondents: include session instructions
We can include session instructions in the email.

### PR DESCRIPTION
**Pull request by HirdayGupta**
Fixes #5768 

Includes Session Instructions in the Feedback Session Opening, Feedback Session Reminder, and Feedback Session Closing emails. It is also included in the email sent to instructors showing an example of the email sent to students.

Also updated expected HTML files using Godmode.

**Screenshot of new email design:**

screen shot 2017-04-16 at 5 35 13 am

**NOTE:** This is a continuation of Pull Request #7070. Apologies for the new PR, the old branch was giving me errors when I updated it from the master (The dev issues that compelled me to close that branch: #7108). 

@leeyimin this implements both the change in `max-width` to 600px and the shift to a table layout for outlook support. Also note, I have made some changes from the styling you had recommended for the table in the old PR. 

Additionally, for the line below the heading, please let me know if you would prefer `border-top` or the existing `<hr/>`.

### CODE REVIEW

**Comment by damithc** [comment]
Good work @HirdayGupta 
While we are at it, why not put the other three session details inside the box too? Wouldn't it look better that way?

**Comment by HirdayGupta** [comment]
Thank you! @damithc @leeyimin 

@damithc something like this?

screen shot 2017-04-16 at 3 53 34 pm

This is the code being displayed:

```HTML
<div>
    <table style="max-width:600px;border:1px solid black;border-collapse:collapse;margin-left:10px;">
        <tr style="">
            <th colspan="2" style="padding:10px;text-align:center;font-size:120%;">
                Feedback Session Details
            </th>
        </tr>
        <tr>
            <td style="padding:10px;border:1px solid black;">
                <strong>
                    Course:
                </strong>
            </td>
            <td style="padding:10px;border:1px solid black;">
                [CS183] Artificial Intelligence
            </td>
        </tr>

        <tr>
            <td style="padding:10px;border:1px solid black;">
                <strong>
                    Feedback Session Name:
                </strong>
            </td>
            <td style="padding:10px;border:1px solid black;">
                Feedback Session 1
            </td>
        </tr>

        <tr>
            <td style="padding:10px;border:1px solid black;">
                <strong>
                    Deadline:
                </strong>
            </td>
            <td style="padding:10px;border:1px solid black;">
                Mon, 01 May 2017, 05:29 AM
            </td>
        </tr>

        <tr>
            <td style="padding:10px;border:1px solid black;">
                <strong>
                    Session Instructions:
                </strong>
            </td>
            <td style="padding:10px;border:1px solid black;">
                <p>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                    Integer pharetra cursus metus vel tristique.
                    Morbi vel dolor maximus, viverra nulla auctor, suscipit dolor.
                    In pretium viverra nibh eget faucibus.
                    Morbi at libero ac orci egestas pharetra sed eu lorem.
                    Ut ac tincidunt nisi. Duis varius condimentum risus eu luctus.
                    Nunc quis lacinia tellus, id posuere felis. Pellentesque massa lacus, consectetur vel consequat vitae, mattis ac mi.
                    Nullam bibendum, nulla aliquam placerat ullamcorper, neque ipsum dignissim eros, in placerat sapien mi ac lorem.
                    Vestibulum a posuere tortor, eu ultrices orci.
                </p>
            </td>
        </tr>
    </table>
</div>
```

**Comment by damithc** [comment]
> Since this particular changes only Feedback Session Opening, Feedback Session Reminder, and Feedback Session Closing emails, and this change will lead to a departure from common formatting, would you suggest I add this table to all emails in this PR itself? Or should we just implement the table for these emails?

Let's do the others as separate PRs. You can do them yourself if you want.

> @damithc something like this?

Yes, looks good.

**Comment by HirdayGupta** [comment]
> Let's do the others as separate PRs. You can do them yourself if you want.

Understood. Will work on it soon.

@damithc @leeyimin requested changes have been made. Ready for review. 

**Screenshot of test email:**

screen shot 2017-04-16 at 4 53 04 pm

File: src/main/resources/userEmailTemplate-feedbackSession.html [hunk 1]
```diff
@@ -4,9 +4,58 @@
...
+                <td style="padding:10px;border:1px solid black;">
+                    <strong>
+                        Session Instructions:
+                    </strong>
+                </td>
+                <td style="padding:10px;border:1px solid black;">
+                    ${sessionInstructions}
+                </td>
+            </tr>
+        </table>
```
Comment (damithc) [inline src/main/resources/userEmailTemplate-feedbackSession.html]: Does the output look very bad if we don't use so much inline styles?

Comment (HirdayGupta) [inline src/main/resources/userEmailTemplate-feedbackSession.html]: Here's how it looks without padding and without border:
screen shot 2017-04-16 at 5 51 39 pm

Here's how it looks with padding and without border:
screen shot 2017-04-16 at 5 52 10 pm

In my opinion, the inline styles on the `<table>` tag (i.e. `max-width`, `border`, `border-collapse` and `margin-left`), the `padding` on all the table elements and the `text-align:center` on the table header are essential. We could get rid of the borders on the rest of the `<td>'s` if you so wish and the `font-size` is just to distinguis the table header from the row headers.

Comment (HirdayGupta) [inline src/main/resources/userEmailTemplate-feedbackSession.html]: We could also, alternatively, get rid of the table header (**Feedback Session Details**) altogether. The email would then look like this(no borders):

screen shot 2017-04-16 at 6 00 40 pm

Comment (damithc) [inline src/main/resources/userEmailTemplate-feedbackSession.html]: Let's use minimal inline styling. It's good enough if the email doesn't look ugly (no need to look pretty). No need for the header.

**Comment by HirdayGupta** [comment]
@damithc Removed some of the unnecessary styling in the latest commits. Kept the padding style for the table cells as I felt content became indistinguishable between rows otherwise(esp when one field, say instructions, is extremely long). Let me know if you think otherwise, and I will remove the padding as well. 

**Below is the screenshot of what the new changes look like:**

screen shot 2017-04-17 at 1 36 58 am

**Here is the problem with removing padding I was talking about:**

**Here it is without any padding:** (The headers of the rows are indistinguishable and confusing) 
screen shot 2017-04-17 at 1 42 38 am

**Here it is with padding:** 
screen shot 2017-04-17 at 1 41 41 am

**Additionally,** because of the long instructions, I was considering adding `vertical-align:top` to the styling for the "Session Instructions" cell (screenshot below). However, since this was an additional style (and only a suggestion to make it look nicer and more comprehensible), I decided to not include it in the commits and wait for your approval to include it. 

screen shot 2017-04-17 at 2 11 58 am

**Comment by damithc** [comment]
We can keep padding and `vertical-align:top `

**Comment by HirdayGupta** [comment]
@damithc added `vertical-align:top`. Ready for review.

**Pull request merged**